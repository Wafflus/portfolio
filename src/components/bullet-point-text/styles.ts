import styled from "styled-components";

export const BulletPointTextStyled = styled.li`
    color: ${({ theme }) => theme.data.text.lightColor};
    font-family: '${({ theme }) => theme.data.fontFamilies.outfit}', sans-serif;
    font-size: 1.125rem;
    letter-spacing: 0.1rem;
    line-height: 120%;
    list-style-type: none;
    position: relative;
    text-align: center;
    transition: transform 0.5s;

    &:hover {
        transform: scale(102%);
    }
    
    ${props => props.theme.breakpoint.above('lg',
        `
            &::before {
                background-color: ${props.theme.data.secondaryColor};
                border-radius: 50%;
                content: '';
                height: 20px;
                left: -40px;
                position: absolute;
                transition: background-color 0.5s;
                width: 20px;
            }

            &:hover::before {
                background-color: ${props.theme.data.primaryColor};
            }
        `
    )}
`;