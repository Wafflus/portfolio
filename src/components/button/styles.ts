import { IButtonStyledProps } from "./types";
import styled, { css } from "styled-components";

export const ButtonStyled = styled.a<IButtonStyledProps>`
    align-items: center;
    background-color: ${({ theme }) => theme.data.secondaryColor};
    border-radius: 8px;
    color: ${({ theme }) => theme.data.text.darkColor};
    display: flex;
    font-size: 2rem;
    font-variant: small-caps;
    gap: 16px;
    height: 66px;
    justify-content: center;
    letter-spacing: 0.1rem;
    max-width: 300px;
    min-width: 200px;
    padding: 8px;
    text-decoration: none;
    transition: transform 0.5s;
    width: 100%;

    ${props => props.outline && css`
        background-color: transparent;
        border: 1px solid ${({ theme }) => theme.data.primaryColor};
        color: #93855B;
    `}

    &:hover {
        box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.1);
        transform: scale(102%);
    }
`;