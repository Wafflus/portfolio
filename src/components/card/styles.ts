import styled from "styled-components";

export const CardStyled = styled.article`
    border-radius: 8px;
    padding: 16px;

    &:nth-child(odd) {
        background-color: ${({ theme }) => theme.data.highlightColor};
    }

    &:hover {
        outline: 3px solid ${({ theme }) => theme.data.borderColor};

        & > div > img {
            border: 3px solid ${({ theme }) => theme.data.primaryColor};
            transform: scale(102%);
        }
    }

    ${props => props.theme.breakpoint.above('lg',
        `
            &:nth-child(even) > div {
                flex-direction: row-reverse;
            }
        `
    )}
`;

export const TitleStyled = styled.h3`
    color: ${({ theme }) => theme.data.text.darkColor};
    font-size: 1.5rem;
    font-variant: small-caps;
    font-weight: 600;
    letter-spacing: 0.1rem;
    text-align: center;
    
    ${props => props.theme.breakpoint.above('lg',
        `
            font-size: 2rem;
            text-align: start;
        `
    )}
`;

export const ToolsStyled = styled.h4`
    color: ${({ theme }) => theme.data.text.defaultColor};
    font-variant: small-caps;
    letter-spacing: 0.1rem;
    text-align: center;
`;

export const DescriptionStyled = styled.p`
    color: ${({ theme }) => theme.data.text.defaultColor};
    flex: 1;
    letter-spacing: 0.1rem;
`;