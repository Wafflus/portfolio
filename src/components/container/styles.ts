import { IContainerStyledProps } from "./types";
import styled from "styled-components";

export const ContainerStyled = styled.div<IContainerStyledProps>`
    height: 100%;
    max-width: 1440px;
    padding: ${props => props.verticalPadding} ${props => props.horizontalPadding};
    width: 100%;
    z-index: 200;
`;