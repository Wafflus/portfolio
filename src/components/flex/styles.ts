import { IFlexStyledProps } from "./types";
import styled from "styled-components";

export const FlexStyled = styled.div<IFlexStyledProps>`
    align-items: ${props => props.defaultFlexProps?.alignItems};
    display: flex;
    flex-direction: ${props => props.defaultFlexProps?.flexDirection};
    gap: ${props => props.defaultFlexProps?.gap};
    justify-content: ${props => props.defaultFlexProps?.justifyContent};
    max-width: ${props => props.defaultFlexProps?.maxWidth};
    width: 100%;
    
    ${props => props.theme.breakpoint.above('lg',
        `
            align-items: ${props.largeBreakpointFlexProps?.alignItems};
            flex-direction: ${props.largeBreakpointFlexProps?.flexDirection};
            gap: ${props.largeBreakpointFlexProps?.gap};
            justify-content: ${props.largeBreakpointFlexProps?.justifyContent};
            max-width: ${props.largeBreakpointFlexProps?.maxWidth};
        `
    )}
`;