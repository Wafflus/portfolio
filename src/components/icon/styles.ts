import styled from "styled-components";

import { IImageProps } from "./types";

export const IconStyled = styled.img<IImageProps>`
    width: ${props => props.size}${props => props.unit};
    height: ${props => props.size}${props => props.unit};
    border-radius: 50%;
`;