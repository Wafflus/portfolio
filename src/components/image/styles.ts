import styled from "styled-components";

export const ImageStyled = styled.img`
    border-radius: 8px;
    border: 3px solid transparent;
    height: 100%;
    max-height: 350px;
    max-width: 500px;
    transition: transform 0.5s, border 0.5s;
    width: 100%;
`;