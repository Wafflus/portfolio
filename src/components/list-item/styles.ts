import { IListItemStyledProps } from "./types";
import styled from "styled-components";

export const ListItemStyled = styled.li<IListItemStyledProps>`
    list-style-type: ${props => props.listStyleType};
    position: relative;
`;