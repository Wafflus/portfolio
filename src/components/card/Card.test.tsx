import "jest-styled-components";
import { breakpoints, theme } from "styles";
import { Card } from ".";
import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import portfolio from 'assets/portfolio.png';

test('card renders', () => {
    const { container } = render(
        <ThemeProvider theme={theme}>
            <Card alt="Image alternative text" description="Card description" src={portfolio} title="Card" tools={['Tool 1', 'Tool 2']} />
        </ThemeProvider>
    );

    expect(container.firstChild).toMatchSnapshot();
});

test('should render github button', () => {
    const card = render(
        <ThemeProvider theme={theme}>
            <Card alt="" description="" src="" title="" tools={[]} links={{ github: 'https://example.com' }}/>
        </ThemeProvider>
    );

    const githubButton = card.getByText(/github/i).closest('a');

    expect(githubButton).toBeInTheDocument();
});

test('should render google play button', () => {
    const card = render(
        <ThemeProvider theme={theme}>
            <Card alt="" description="" src="" title="" tools={[]} links={{ googlePlay: 'https://example.com' }}/>
        </ThemeProvider>
    );

    const googlePlayButton = card.getByText(/google play/i).closest('a');

    expect(googlePlayButton).toBeInTheDocument();
});

test('should render website button', () => {
    const card = render(
        <ThemeProvider theme={theme}>
            <Card alt="" description="" src="" title="" tools={[]} links={{ website: 'https://example.com' }}/>
        </ThemeProvider>
    );

    const websiteButton = card.getByText(/live/i).closest('a');

    expect(websiteButton).toBeInTheDocument();
});

test('should render youtube button', () => {
    const card = render(
        <ThemeProvider theme={theme}>
            <Card alt="" description="" src="" title="" tools={[]} links={{ youtube: 'https://example.com' }}/>
        </ThemeProvider>
    );

    const youtubeButton = card.getByText(/youtube/i).closest('a');

    expect(youtubeButton).toBeInTheDocument();
});

test('should render an image', () => {
    const card = render(
        <ThemeProvider theme={theme}>
            <Card alt="Alternative text." description="" src={portfolio} title="" tools={[]}/>
        </ThemeProvider>
    );

    const githubButton = card.getByAltText('Alternative text.').closest('img');

    expect(githubButton).toBeInTheDocument();
});

test('should render a title', () => {
    const card = render(
        <ThemeProvider theme={theme}>
            <Card alt="" description="" src={portfolio} title="Title" tools={[]}/>
        </ThemeProvider>
    );

    const githubButton = card.getByText('Title').closest('h3');

    expect(githubButton).toBeInTheDocument();
});

test('should render the tools', () => {
    const card = render(
        <ThemeProvider theme={theme}>
            <Card alt="" description="" src={portfolio} title="" tools={['Tool 1', 'Tool 2']}/>
        </ThemeProvider>
    );

    const githubButton = card.getByText('Tool 1, Tool 2').closest('h4');

    expect(githubButton).toBeInTheDocument();
});

test('should render a description', () => {
    const card = render(
        <ThemeProvider theme={theme}>
            <Card alt="" description="Description" src={portfolio} title="" tools={[]}/>
        </ThemeProvider>
    );

    const githubButton = card.getByText('Description').closest('p');

    expect(githubButton).toBeInTheDocument();
});

test('should have outline on hover', () => {
    const { container } = render(
        <ThemeProvider theme={theme}>
            <Card alt="" description="" src={portfolio} title="" tools={[]}/>
        </ThemeProvider>
    );

    expect(container.firstChild).toHaveStyleRule('outline', `3px solid ${theme.data.borderColor}`, {
        modifier: ':hover'
    });
});

test('should scale inside image on hover', () => {
    const { container } = render(
        <ThemeProvider theme={theme}>
            <Card alt="" description="" src={portfolio} title="" tools={[]}/>
        </ThemeProvider>
    );

    expect(container.firstChild).toHaveStyleRule('transform', 'scale(102%)', {
        modifier: ':hover > div > img'
    });
});

test('should give a border to the image inside on hover', () => {
    const { container } = render(
        <ThemeProvider theme={theme}>
            <Card alt="" description="" src={portfolio} title="" tools={[]}/>
        </ThemeProvider>
    );

    expect(container.firstChild).toHaveStyleRule('border', `3px solid ${theme.data.primaryColor}`, {
        modifier: ':hover > div > img'
    });
});

test('should change odd children background color', () => {
    const { container } = render(
        <ThemeProvider theme={theme}>
            <Card alt="" description="" src={portfolio} title="" tools={[]}/>
        </ThemeProvider>
    );

    expect(container.firstChild).toHaveStyleRule('background-color', theme.data.highlightColor, {
        modifier: ':nth-child(odd)'
    });
});

test('should reverse even children on large (lg) screens', () => {
    const { container } = render(
        <ThemeProvider theme={theme}>
            <Card alt="" description="" src={portfolio} title="" tools={[]}/>
        </ThemeProvider>
    );

    expect(container.firstChild).toHaveStyleRule('flex-direction', 'row-reverse', {
        media: `(min-width: ${breakpoints.lg}px)`,
        modifier: ':nth-child(even) > div'
    });
});