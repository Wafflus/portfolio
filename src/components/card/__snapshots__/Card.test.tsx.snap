// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card renders 1`] = `
.c0 {
  border-radius: 8px;
  padding: 16px;
}

.c0:nth-child(odd) {
  background-color: #FFFAE9;
}

.c0:hover {
  outline: 3px solid #FFF0BF;
}

.c0:hover > div > img {
  border: 3px solid #F9D564;
  -webkit-transform: scale(102%);
  -ms-transform: scale(102%);
  transform: scale(102%);
}

.c5 {
  color: #756A47;
  font-size: 1.5rem;
  font-variant: small-caps;
  font-weight: 600;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
  text-align: center;
}

.c6 {
  color: #93855B;
  font-variant: small-caps;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
  text-align: center;
}

.c7 {
  color: #93855B;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 24px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  max-width: 100%;
  width: 100%;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 16px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  max-width: 100%;
  width: 100%;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 8px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  max-width: 100%;
  width: 100%;
}

.c8 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 16px;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  max-width: 100%;
  width: 100%;
}

.c2 {
  border-radius: 8px;
  border: 3px solid transparent;
  height: 100%;
  max-height: 350px;
  max-width: 500px;
  -webkit-transition: -webkit-transform 0.5s,border 0.5s;
  -webkit-transition: transform 0.5s,border 0.5s;
  transition: transform 0.5s,border 0.5s;
  width: 100%;
}

@media (min-width:992px) {
  .c0:nth-child(even) > div {
    -webkit-flex-direction: row-reverse;
    -ms-flex-direction: row-reverse;
    flex-direction: row-reverse;
  }
}

@media (min-width:992px) {
  .c5 {
    font-size: 2rem;
    text-align: start;
  }
}

@media (min-width:992px) {
  .c1 {
    -webkit-align-items: stretch;
    -webkit-box-align: stretch;
    -ms-flex-align: stretch;
    align-items: stretch;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    gap: 24px;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    max-width: 100%;
  }
}

@media (min-width:992px) {
  .c3 {
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    gap: 16px;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    max-width: 100%;
  }
}

@media (min-width:992px) {
  .c4 {
    -webkit-align-items: start;
    -webkit-box-align: start;
    -ms-flex-align: start;
    align-items: start;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    gap: 8px;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    max-width: 100%;
  }
}

@media (min-width:992px) {
  .c8 {
    -webkit-align-items: start;
    -webkit-box-align: start;
    -ms-flex-align: start;
    align-items: start;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    gap: 16px;
    -webkit-box-pack: start;
    -webkit-justify-content: start;
    -ms-flex-pack: start;
    justify-content: start;
    max-width: 100%;
  }
}

<article
  class="c0"
>
  <div
    class="c1"
  >
    <img
      alt="Image alternative text"
      class="c2"
      src="portfolio.png"
    />
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <h3
          class="c5"
        >
          Card
        </h3>
        <h4
          class="c6"
        >
          Tool 1, Tool 2
        </h4>
      </div>
      <p
        class="c7"
      >
        Card description
      </p>
      <div
        class="c8"
      />
    </div>
  </div>
</article>
`;
